src/com/xilinx/rapidwright/timing/TimingModel.java

```diff
@@ -500,17 +502,82 @@ protected List<TimingGroup> determineGroups(List<Node> nodes, List<IntentCode> n
         return result;
     }
 
+    private Pair<Integer,Integer> findReferenceTileLocation(Device d) {
+        class Helper {

```

Comment:
Perhaps consider a more descriptive name than `Helper`?  

The Guava library has a map that mostly does what `Helper` does:
```java
        SetMultimap<Integer, Integer> map = TreeMultimap.create();
        map.put(1, 2);
        map.put(1, 1);
        map.put(1, 1);
        System.out.println(map.get(1));
        // output is [1,2]
```
RapidWright already includes Guava, so it would be a (mostly) drop in replacement.  Ultimately, it would be nice to not have an inner class.